godot-core/src/task/futures.rs

```diff
@@ -11,26 +11,38 @@ use std::future::{Future, IntoFuture};
 use std::pin::Pin;
 use std::sync::{Arc, Mutex};
 use std::task::{Context, Poll, Waker};
-
-use crate::builtin::{Callable, RustCallable, Signal, Variant};
+use std::thread::ThreadId;
+
+use crate::builtin::{
+    Aabb, Basis, Callable, Color, Dictionary, GString, NodePath, PackedByteArray, PackedColorArray,
+    PackedFloat32Array, PackedFloat64Array, PackedInt32Array, PackedInt64Array, PackedStringArray,
+    PackedVector2Array, PackedVector3Array, PackedVector4Array, Plane, Projection, Quaternion,
+    Rect2, Rect2i, Rid, RustCallable, Signal, StringName, Transform2D, Transform3D, Variant,
+    VariantArray, Vector2, Vector2Axis, Vector2i, Vector3, Vector3Axis, Vector3i, Vector4,
+    Vector4i,
+};
```

Comment:
Instead of importing them all here, could you qualify them in the macro as `$crate::builtin::`? That way, you'd also solve the CI failure on 4.2 where `PackedVector4Array` is missing.

But maybe check some of my other comments as well.